packages/grpc-native-core/src/client.js

```diff
@@ -500,21 +499,27 @@ exports.Client = Client;
 Client.prototype.makeUnaryRequest = function(method, serialize, deserialize,
                                              argument, metadata, options,
                                              callback) {
-  /* While the arguments are listed in the function signature, those variables
-   * are not used directly. Instead, ArgueJS processes the arguments
-   * object. This allows for simple handling of optional arguments in the
-   * middle of the argument list, and also provides type checking. */
-  var args = arguejs({method: String, serialize: Function,
-                      deserialize: Function,
-                      argument: null, metadata: [Metadata, new Metadata()],
-                      options: [Object], callback: Function}, arguments);
-  var call = getCall(this.$channel, method, args.options);
+  if (!(metadata instanceof Metadata)) {
+    callback = options;
+    options = metadata;
+    metadata = new Metadata();
+  }
+  if (options instanceof Function) {
+    callback = options;
+    options = {};
+  }
+  if (!((metadata instanceof Metadata) &&
+	(options instanceof Object) &&
```

Comment:
This causes requests to throw when `options === undefined`. If this is new, intended behavior it is a breaking change from argue.js's behavior. This caused quite the headache for me and my team, as we didn't think to look to the grpc lib originally, because we felt such a breaking change would not be in a minor version bump.

For reference, we were calling the client request as such:
```
function makeCall(rpc, message, meta, options) {
  return new Bluebird((resolve, reject) => {
    client[rpc](message, meta, options, (err, res) => {
        if (err) {
          return reject(err);
        }
        return resolve(res);
    });
  });
}
```

Giving the `options` argument a default of `= {}` resolved the issue, but I don't feel like this should be necessary.

Reply:
Thank you for reporting this issue. It was also reported in #178 and a fix is in #179. We will try to publish a patch release soon after that is merged.